synapse/python_dependencies.py

```diff
@@ -75,6 +75,7 @@
     "attrs>=19.2.0,!=21.1.0",
     "netaddr>=0.7.18",
     "Jinja2>=2.9",
+    "MarkupSafe<=3.0.0",
```

Comment:
I'm curious where the 3.0.0 version came from (and why less than, instead of greater than some package number).

Reply:
Where did the 2.1.1 come from though? Is that what Jinja2 was pinning at?

Reply:
That's just the latest version of `MarkupSafe`. Jinja2 uses `MarkupSafe>=2.0`: https://github.com/pallets/jinja/blob/e542e100027aaa131df73ad06b71ba9a4ee348c8/setup.py#L6

Reply:
for reference, if your goal is to pin to `2.x`, you probably want `~=2.0`. See https://peps.python.org/pep-0440/#compatible-release.

Reply:
The goal from my POV was to not change what we were already pulling in, but making it an explicit dependency.